### PR TITLE
Fix FilmGrain2 visual crosshatching pattern

### PR DESCRIPTION
**Pull request by Alo81**

### CODE REVIEW

File: Shaders/FilmGrain2.fx [hunk 1]
```diff
@@ -37,7 +37,7 @@ uniform float timer < source = "timer"; >;
 float4 rnm(in float2 tc)
 {
 	// A random texture generator, but you can also use a pre-computed perturbation texture
-	float noise = sin(dot(tc + timer.xx * 0.5, float2(12.9898, 78.233))) * 43758.5453;
+	float noise = sin(dot(float3(tc.x, tc.y, timer), float2(12.9898, 78.233))) * 43758.5453;
```
Comment (crosire) [inline Shaders/FilmGrain2.fx]: Somethings looks off here. A dot product between a `float3` and `float2` doesn't make sense.

Comment (Alo81) [reply Shaders/FilmGrain2.fx]: For additional context that may help, here is the original line, from Martins foundational shader.  
    float noise =  sin(dot(tc + vec2(timer,timer),vec2(12.9898,78.233))) * 43758.5453;

Realistically, when I originally ported this over was more of a "Change variable types until it runs and looks right" endeavor.  The new line is how it was how I had originally ported it into GeDoSaTo, then over to ReShade.  When the update was made to it which changed that line is when the noise showed noticeable crosshatching.  Changing it back removed it.  I don't have a greater understanding of the original intent of that line.  

I made up a comparison screenshot.  I looked at the original shader file from Martin and reimplemented it for the screenshot.  The original line of code reimplemented as:
    float noise =  sin(dot(tc + float2(timer,timer),float2(12.9898,78.233))) * 43758.5453;
(View image at full size to best see the patterns)
film grain comparisons

Visually, this fix seems to look most appropriate, but the reason the others don't might have some other underlying reason.

Comment (crosire) [reply Shaders/FilmGrain2.fx]: But unless I'm misreading something here, the new line is now equivalent to:
```hlsl
float noise = sin(dot(tc, float2(12.9898, 78.233))) * 43758.5453;
```
This means the timer value is ignored entirely and the noise is static now?

Comment (Alo81) [reply Shaders/FilmGrain2.fx]: The noise isn't static, and moves the way you'd naturally expect noise to.  Testing in practice, the timer variable utilized in 
float2 rotCoordsR = coordRot(texCoord, timer + rotOffset.x);
is what appears to end up generating the actual noise movement.  Removing timer from the above does cause the noise to become completely static. 

Yeah, you're right that the two are functionally identical, I just tested as well to verify. 

There may have been a different intent originally for the timer in creation of the noise variables value, but it seems in the use cases above, all the timer in the noise generation does is end up creating a visible pattern.

Comment (crosire) [reply Shaders/FilmGrain2.fx]: Alright, would you mind simplifying it to the code I posted? Will merge then.

Comment (Alo81) [reply Shaders/FilmGrain2.fx]: Yep no problem.  Should be done now.

**Pull request merged**